Pocket/pocket-ios
# In 1504: Readable Width for Saves and Search

## Summary
Adds readable width to Saves and mimics it for Search.

## References 
https://getpocket.atlassian.net/browse/IN-1504

## Implementation Details
Saves is built on UIKit and so we are able to add `section.contentInsetsReference = .readableContent` to the content cells layout config.

Search is built using SwiftUI, which does not appear to have a readableContent setting.
Therefore, if the interface width size class is .regular we restrict the list width to a preset amount (currently 500 points)

## Test Steps
Launch and use Saves and Search as normal.
On iPad you may then enable split screen and while changing the window size you will notice the views adapt to their width.
iPhones do not have a landscape layout for Search and Saves, only Reader which already uses Readable Width.

## PR Checklist:

## Screenshots
![Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-05-31 at 09 46 23](https://github.com/Pocket/pocket-ios/assets/8590135/7a10921d-3642-40ff-9e2e-641bf0051b4e)
![Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-05-31 at 09 46 17](https://github.com/Pocket/pocket-ios/assets/8590135/4bcc5905-b3a6-4959-b219-b309344d2f5f)
![Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-05-31 at 09 46 12](https://github.com/Pocket/pocket-ios/assets/8590135/29b9de74-1e0b-4658-b3da-fe1ead11f4c2)
![Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-05-31 at 09 46 04](https://github.com/Pocket/pocket-ios/assets/8590135/d996ff64-8209-4b0b-9340-a892ab9bc5d4)
![Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-05-31 at 09 45 59](https://github.com/Pocket/pocket-ios/assets/8590135/d16ed730-4627-4283-a3f7-de9833cecfa5)
![Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-05-31 at 09 45 51](https://github.com/Pocket/pocket-ios/assets/8590135/e97ad0f3-93e9-4722-bc90-0613b3191ee3)
